modules/ROOT/pages/apikit-odatav4-expand-feature.adoc

```diff
@@ -0,0 +1,98 @@
+* The `Managers` entity set is a collection of `Manager` entities.
+* The API location is `https://example.com/api/`.
+
+Given this setup, the following requests retrieve employees and their managers:
+
+* Use `GET https://example.com/api/Employees` to retrieve a list of employees.
+* Gather the `Manager` IDs through the `ReferentialConstraint` of their relationship.
+* Use `GET https://example.com/api/Managers?$filter=ManagerID in (id1, id2, ..., idN)` to retrieve a list of related managers.
+* Write a small program to merge the lists based on join criteria.
+
+If using the `$expand` feature, requests are `GET https://example.com/api/Employees?$expand=Manager` to retrieve a list of employees with their `Manager` navigation property expanded.
+
+== The `generic expand` Feature
+
+The `generic expand` feature enables you to reuse existing flow implementations as a starting point for expanding requests. Using this feature, you can focus on the relevant data retrieval operations while the module performs the necessary steps to ensure that the `$expand` fields complete.
+
+An application with the `generic expand` feature enabled works as any other OData Mule application. However, during the flow execution, once the module retrieves information for the main entity set, it performs additional internal requests gathering the information for any `expand` field. If there are errors during the expansion operation, the consumer can configure the final behavior, or the consumer can ignore it with a `null` value.
+
+== The `generic expand` Operation
+
+The `generic expand` feature dispatches additional queries to the application when the original flow completes. Because this is done based on the application `EDM`, you must make extra annotations to add enough context to the module:
+
+. The `EntitySet` where the expansion takes place has a `NavigationPropertyBinding` for every expandable `NavigationProperty` of its `Entity`. +
+This is used to select which `<entity-collection-source/>` is executed when expanding that property.
+. Each expandable `NavigationProperty` has a `ReferentialConstraint` attached. +
+This creates the inner query, where the value of `Property` is matched against the value of `ReferencedProperty`.
+
+=== Flow Execution
+
+. The main flow is executed and completed resulting in a collection of `entities`.
+. The OData module gathers all the properties that need expansion.
+. For each `NavigationProperty`:
+.. Get the `source property` from its `ReferentialConstraint`.
+.. Get the `referenced property` from its `ReferentialConstraint`.
+.. Get the `destination entity set` from its `NavigationPropertyBinding`.
+.. For each of the `entities` get the resolved `source property`. Add these values to a `source property list`.
+.. Execute the `internal` request by invoking the `<entity-collection-source/>` for `destination entity set` with `$filter=referenced property` in `(...source property list)`.
```

Comment:
I don't think internal has to be highlighted here. It's not a specification thing nor a name for something that will be referenced later.

Reply:
OK! Thanks for the clarification!